src/plugin.ts

```diff
@@ -212,11 +212,43 @@ export function openGraphImagePlugin(options: Options): Plugin {
           const optimizedImageBuffer = await optimizeImageBuffer.toBuffer()
           const imageName = `${data.name}.${format}`
 
+          const outputPath = await fromOutputDirectory(imageName)
+
           images.push({
             name: imageName,
-            path: await fromOutputDirectory(imageName),
+            path: outputPath,
             content: optimizedImageBuffer,
           })
+
+          // check if outputPath is somewhere in `public` directory
+          if (outputPath.includes("public")) {
+            // strip the path upto the public part of the path
+            const imagePath = outputPath.split("public")[1];
+
+            // this property does exist in the remixConfig
+            // but are missing in the types
+            // not sure if this is the best way to check for it
+            // TODO: there must be a better way to resolve this
+            if (
+              "buildDirectory" in remixContext.remixConfig &&
+              typeof remixContext.remixConfig.buildDirectory === "string"
+            ) {
+              if (imagePath) {
+                const clientPath = path.join(
+                  remixContext.remixConfig.buildDirectory,
+                  "client",
+                  imagePath
+                );
+
+                images.push({
+                  name: imageName,
+                  path: clientPath,
+                  content: optimizedImageBuffer,
+                });
```

Comment:
This entire thing feels a bit "hacky" and might be missing handling the edge cases